cloudbreak/config/config.go

```diff
@@ -89,54 +85,24 @@ func configRead(c *cli.Context) error {
 	if len(output) == 0 {
 		set(fl.FlOutputOptional.Name, config.Output)
 	}
-
-	if len(authType) == 0 {
-		authType = config.AuthType
-		if len(authType) == 0 {
-			set(fl.FlAuthTypeOptional.Name, common.OAUTH2)
-		} else {
-			if authType != common.OAUTH2 && authType != common.BASIC {
-				utils.LogErrorAndExit(errors.New(fmt.Sprintf("invalid authentication type, accepted values: [%s, %s]", common.OAUTH2, common.BASIC)))
-			}
-			set(fl.FlAuthTypeOptional.Name, authType)
-		}
-	}
 	PrintConfig(*config)
 	if len(server) == 0 {
 		set(fl.FlServerOptional.Name, config.Server)
 	}
-	if len(username) == 0 {
-		set(fl.FlUsername.Name, config.Username)
-	}
-	if len(password) == 0 {
-		if len(config.Password) == 0 {
-			fmt.Println("Enter Password: ") // needs trailing new-line
-			bytePassword, ioErr := terminal.ReadPassword(int(syscall.Stdin))
-			fmt.Println()
-			if ioErr != nil {
-				utils.LogErrorAndExit(ioErr)
+	if len(refreshToken) == 0 {
+		if len(config.RefreshToken) == 0 {
+			token := caasauth.NewRefreshToken(c.String(fl.FlServerOptional.Name))
+			err = WriteConfigToFile(GetHomeDirectory(), config.Server, config.Output, profile, config.Workspace, token)
+			if err != nil {
+				utils.LogErrorAndExit(err)
 			}
-			set(fl.FlPassword.Name, string(bytePassword))
+			set(fl.FlRefreshTokenOptional.Name, token)
 		} else {
-			set(fl.FlPassword.Name, config.Password)
+			set(fl.FlRefreshTokenOptional.Name, config.RefreshToken)
 		}
 	}
 	if len(workspace) == 0 {
-		if len(config.Workspace) == 0 {
-			workspaceList := ws.GetWorkspaceList(c)
-			var workspaceID string
-			for _, workspace := range workspaceList {
-				if workspace.Name == c.String(fl.FlUsername.Name) {
-					workspaceID = strconv.FormatInt(workspace.ID, 10)
-				}
-			}
-
-			err = WriteConfigToFile(GetHomeDirectory(), config.Server, config.Username, config.Password, config.Output, profile, config.AuthType, config.Username)
-			if err != nil {
-				utils.LogErrorAndExit(err)
-			}
-			set(fl.FlWorkspaceOptional.Name, workspaceID)
-		} else {
+		if len(config.Workspace) != 0 {
```

Comment:
Should be merged with the previous if statement

Reply:
ok